nevergrad/optimization/oneshot.py

```diff
@@ -87,6 +88,14 @@ class OneShotOptimizer(base.Optimizer):
     # pylint: disable=abstract-method
     one_shot = True
 
+    def _internal_ask_candidate(self) -> p.Parameter:
+        out = self.parametrization.spawn_child()
+        with p.helpers.deterministic_sampling(out):
+            # the values will be sampled deterministically since there is no reason not to
+            # with one shot algorithms
+            out.set_standardized_data(self._internal_ask())
+        return out
```

Comment:
@teytaud for one shot algorithms, the sampling of choice parameters is performed deterministically. I feel that makes much more sense, but maybe I am mistaken?

Reply:
I agree with you!